build-support/run_clang_tidy_extra.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+"""
+A helper class, to suppress execution of clang-tidy.
+
+In clang-tidy-6.0, if the clang-tidy configuration file suppresses ALL checks,
+(e.g. via a .clang-tidy file), clang-tidy will print usage information and
+exit with a return code of 0. Harmless but verbose. In later versions of 
+clang-tidy the return code becomes 1, making this a bigger problem.
+
+This helper addresses the problem by suppressing execution according to
+the configuration in this file.
+"""
+
+import re
+
+class CheckConfig(object):
+    """ Check paths against the built-in config """
+
+    def __init__(self):
+        self._init_config()
+        # debug prints
+        self.debug = False
+        return
+
+    def _init_config(self):
+        """ Any path matching one of the ignore_pats regular expressions,
+            denotes that we do NOT want to run clang-tidy on that item.
+        """
+        self.ignore_pats = [".*/third_party/libpg_query/.*", ]
```

Comment:
Should we make it all of third_party right now, just preemptively?

Reply:
Sure.